plugins/osdn/registry.go

```diff
@@ -145,12 +149,14 @@ func (registry *Registry) WatchPods() error {
 		}
 		pod := obj.(*kapi.Pod)
 
+		registry.podTrackingLock.Lock()
 		switch eventType {
 		case watch.Added, watch.Modified:
 			registry.trackPod(pod)
 		case watch.Deleted:
 			registry.unTrackPod(pod)
 		}
+		registry.podTrackingLock.Unlock()
```

Comment:
Currently this is not an issue but there could be potential issues later on with this approach:
- Since we are taking lock at a wider scope than needed, if a method (trackPod/unTrackPod...) panics for some reason holding the lock, all dependent methods will be blocked
- We need to sprinkle these lock/unlock whenever we access podsByIP

Instead I prefer adding these 3 methods:
GetPodInfo() {
get lock
fetch from podsByIP
release lock
}
SetPodInfo(podInfo) {
get lock
add/set pod info
release lock
}
UnSetPodInfo(podInfo) {
get lock
remove from podsByIP
release lock
}
Use these methods where ever required : in endpointsUpdate, trackPod and untrackPod